frontend/src/metabase/common/components/EntityPicker/hooks/use-log-recent-item.tsx

```diff
@@ -0,0 +1,27 @@
+import { useCallback } from "react";
+
+import { useLogRecentItemMutation } from "metabase/api";
+import { isLoggableActivityModel } from "metabase-types/api";
+
+import type { CollectionPickerItem } from "../../CollectionPicker";
+import type { NotebookDataPickerValueItem } from "../../DataPicker";
+
+export const useLogRecentItem = () => {
+  const [logRecentItem] = useLogRecentItemMutation();
+
+  const handleLogRecentItem = useCallback(
+    (item: CollectionPickerItem | NotebookDataPickerValueItem) => {
+      if (item && isLoggableActivityModel(item)) {
+        logRecentItem({
+          model_id: item.id,
+          model: item.model,
+        });
+      }
+    },
+    [logRecentItem],
+  );
+
+  return {
+    handleLogRecentItem,
```

Comment:
It's unusual for a hook to return a function prefixed with `handle`.

How about `tryLogRecentItem`?